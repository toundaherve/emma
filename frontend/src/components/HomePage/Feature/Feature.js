import React from "react";
import PropTypes from "prop-types";

const Feature = ({ direction }) => {
  return (
    <div class={`featured-item grid-layout ${direction}`}>
      <div class="section-content image-content">
        <picture>
          {/*<!--[if IE 9]>*/}
          <video style={{ display: "none" }}>
            {/*<![endif]-->*/}
            <source
              media="(max-width: 479px)"
              srcset="https://images.prismic.io/brinker-chilis/e7cbde66-69e8-4f99-84eb-9a8ef4eef766_CHI-21748_GCLandingPage_GCinBulk3.jpg?auto=compress,format&amp;rect=0,0,800,600&amp;w=480&amp;h=360"
            />
            <source
              media="(max-width: 991px)"
              srcset="https://images.prismic.io/brinker-chilis/cad8a972-13e0-4677-94d4-294ad4df0778_CHI-21748_GCLandingPage_Mobile-Tablet_GCinBulk3.jpg?auto=compress,format&amp;rect=60,0,360,360&amp;w=496&amp;h=496"
            />
            {/*<!--[if IE 9]>*/}
          </video>
          {/*<![endif]-->*/}
          <img
            class="background-image-blur background-image-center b-lazy unblur b-loaded"
            alt="Chili's Gift Cards Buy in Bulk "
            src="https://images.prismic.io/brinker-chilis/e7cbde66-69e8-4f99-84eb-9a8ef4eef766_CHI-21748_GCLandingPage_GCinBulk3.jpg?auto=compress,format&amp;rect=0,0,800,600&amp;w=800&amp;h=600"
          />
        </picture>
      </div>
      <div class="section-content text-content">
        <div class="text-container">
          <h2>The gift that 'sizzles'</h2>
          <div
            class="featured-description-container"
            style={{ textAlign: "center" }}
          >
            <p>
              Purchase a Chili's gift card for every &amp; any occasion. Our
              e-gift cards are also a perfect way to send a 'thank you' or
              'thinking of you' note from afar.
            </p>
          </div>
          <div class="cta-component text-center">
            <a class="btn" id="home-2-button-1" href="/gift-cards">
              Buy Now
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};

// Feature.propTypes = {
//   direction: PropTypes.oneOf(["left, right"]),
// };

export default Feature;
