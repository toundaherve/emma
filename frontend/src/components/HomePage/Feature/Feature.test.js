import React from "react";
import renderer from "react-test-renderer";

import Feature from "./Feature";

it("renders correctly to the left", () => {
  const component = renderer.create(<Feature direction="left" />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders correctly to the right", () => {
  const component = renderer.create(<Feature direction="right" />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
