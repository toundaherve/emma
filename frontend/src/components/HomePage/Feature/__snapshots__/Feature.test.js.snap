// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly to the left 1`] = `
<div
  class="featured-item grid-layout left"
>
  <div
    class="section-content image-content"
  >
    <picture>
      <video
        style={
          Object {
            "display": "none",
          }
        }
      >
        <source
          media="(max-width: 479px)"
          srcset="https://images.prismic.io/brinker-chilis/e7cbde66-69e8-4f99-84eb-9a8ef4eef766_CHI-21748_GCLandingPage_GCinBulk3.jpg?auto=compress,format&rect=0,0,800,600&w=480&h=360"
        />
        <source
          media="(max-width: 991px)"
          srcset="https://images.prismic.io/brinker-chilis/cad8a972-13e0-4677-94d4-294ad4df0778_CHI-21748_GCLandingPage_Mobile-Tablet_GCinBulk3.jpg?auto=compress,format&rect=60,0,360,360&w=496&h=496"
        />
      </video>
      <img
        alt="Chili's Gift Cards Buy in Bulk "
        class="background-image-blur background-image-center b-lazy unblur b-loaded"
        src="https://images.prismic.io/brinker-chilis/e7cbde66-69e8-4f99-84eb-9a8ef4eef766_CHI-21748_GCLandingPage_GCinBulk3.jpg?auto=compress,format&rect=0,0,800,600&w=800&h=600"
      />
    </picture>
  </div>
  <div
    class="section-content text-content"
  >
    <div
      class="text-container"
    >
      <h2>
        The gift that 'sizzles'
      </h2>
      <div
        class="featured-description-container"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <p>
          Purchase a Chili's gift card for every & any occasion. Our e-gift cards are also a perfect way to send a 'thank you' or 'thinking of you' note from afar.
        </p>
      </div>
      <div
        class="cta-component text-center"
      >
        <a
          class="btn"
          href="/gift-cards"
          id="home-2-button-1"
        >
          Buy Now
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly to the right 1`] = `
<div
  class="featured-item grid-layout right"
>
  <div
    class="section-content image-content"
  >
    <picture>
      <video
        style={
          Object {
            "display": "none",
          }
        }
      >
        <source
          media="(max-width: 479px)"
          srcset="https://images.prismic.io/brinker-chilis/e7cbde66-69e8-4f99-84eb-9a8ef4eef766_CHI-21748_GCLandingPage_GCinBulk3.jpg?auto=compress,format&rect=0,0,800,600&w=480&h=360"
        />
        <source
          media="(max-width: 991px)"
          srcset="https://images.prismic.io/brinker-chilis/cad8a972-13e0-4677-94d4-294ad4df0778_CHI-21748_GCLandingPage_Mobile-Tablet_GCinBulk3.jpg?auto=compress,format&rect=60,0,360,360&w=496&h=496"
        />
      </video>
      <img
        alt="Chili's Gift Cards Buy in Bulk "
        class="background-image-blur background-image-center b-lazy unblur b-loaded"
        src="https://images.prismic.io/brinker-chilis/e7cbde66-69e8-4f99-84eb-9a8ef4eef766_CHI-21748_GCLandingPage_GCinBulk3.jpg?auto=compress,format&rect=0,0,800,600&w=800&h=600"
      />
    </picture>
  </div>
  <div
    class="section-content text-content"
  >
    <div
      class="text-container"
    >
      <h2>
        The gift that 'sizzles'
      </h2>
      <div
        class="featured-description-container"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <p>
          Purchase a Chili's gift card for every & any occasion. Our e-gift cards are also a perfect way to send a 'thank you' or 'thinking of you' note from afar.
        </p>
      </div>
      <div
        class="cta-component text-center"
      >
        <a
          class="btn"
          href="/gift-cards"
          id="home-2-button-1"
        >
          Buy Now
        </a>
      </div>
    </div>
  </div>
</div>
`;
